slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c

```diff
@@ -1681,16 +1875,39 @@ int getBebFPGATemp() {
 #endif
 }
 
-int activate(int enable) {
+int setActivate(int enable) {
+    if (enable < 0) {
+        LOG(logERROR, ("Invalid activate argument: %d\n", enable));
+        return FAIL;
+    }
+    enable = enable == 0 ? 0 : 1;
 #ifdef VIRTUAL
-    if (enable >= 0)
-        eiger_virtual_activate = enable;
-    return eiger_virtual_activate;
+    eiger_virtual_activate = enable;
 #else
-    int ret = Beb_Activate(enable);
-    Feb_Control_activate(ret);
-    return ret;
+    if (!Beb_SetActivate(enable)) {
+        return FAIL;
+    }
+    Feb_Control_activate(enable);
 #endif
+    if (enable == 0) {
```

Comment:
I would switch the  statement around and do if(enabled)

Reply:
done